src/index.js

```diff
@@ -48,49 +37,79 @@ class Notification {
    */
   send(event, notification) {
     const user = notification.to;
-    this.io.sockets[user].emit(event, notification, (ack) => {
-      if (ack) {
-        notificationController
-        .update(notification._id, { sent: true }, //eslint-disable-line
-          (err) => {
-            if (err) throw err;
-          });
-      }
+    const notificationUpdate = new Promise((resolve, reject) => {
+      this.io.sockets[user].emit(event, notification, (ack) => {
+        if (ack) {
+          this.notificationController
+            .update(notification._id, { sent: true }) //eslint-disable-line
+            .then(updatedNotification => resolve(updatedNotification))
+            .catch((err) => {
+              reject(err);
+            });
+        } else {
+          resolve();
+        }
+      });
     });
+    return notificationUpdate;
   }
 
   /**
    * Sends all the unread notifications by user id
    * @param  {string} userId user identifier
    */
   sendUnread(userId) {
-    const self = this;
     if (userId) {
-      notificationController
-      .getUnread(userId, (err, data) => {
-        if (err) throw err;
-        data.forEach((notification) => {
-          self.send('unread', notification);
-        });
+      const notificatonsSent = new Promise((resolve, reject) => {
```

Comment:
I've added promises but I don't know if this will be the default approach. If I recall correctly we wanted to support both `promises` and `callbacks`